src/App.tsx

```diff
@@ -1,62 +1,139 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/bulma.sass';
 import '@fortawesome/fontawesome-free/css/all.css';
 import './App.scss';
 
-import classNames from 'classnames';
 import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
+import { User } from './types/User';
+import { Post } from './types/Post';
+import { getUsers, getPosts } from './api';
 import { Loader } from './components/Loader';
 
+enum Error {
+  None,
+  PostsLoadingError,
+  NoPostsError,
+}
+
 export const App: React.FC = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [selectedUserId, setSelectedUserId] = useState<number | null>(null);
+  const [openedPostId, setOpenedPostId] = useState<number | null>(null);
+  const [openedPost, setOpenedPost] = useState<Post | null>(null);
```

Comment:
why do you need these two states? They always change synchronously. Move `openedPost `out the state